tools/src/main/java/org/apache/rocketmq/tools/command/broker/GetMasterSlaveDiffCommand.java

```diff
@@ -0,0 +1,151 @@
+    public Options buildCommandlineOptions(final Options options) {
+        Option opt = new Option("b", "brokerName", true, "Broker Name");
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option("c", "clusterName", true, "Cluster Name");
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        return options;
+    }
+
+    @Override
+    public void execute(final CommandLine commandLine, final Options options,
+        final RPCHook rpcHook) throws SubCommandException {
+        DefaultMQAdminExt defaultMQAdminExt = new DefaultMQAdminExt(rpcHook);
+        defaultMQAdminExt.setInstanceName(Long.toString(System.currentTimeMillis()));
+        try {
+            if (commandLine.hasOption('b')) {
+                String brokerAddr = commandLine.getOptionValue('b').trim();
+                defaultMQAdminExt.start();
+                printDiffForBroker(defaultMQAdminExt, brokerAddr);
+            } else if (commandLine.hasOption('c')) {
+                String clusterName = commandLine.getOptionValue('c').trim();
+                defaultMQAdminExt.start();
+                printDiffForCluster(defaultMQAdminExt, clusterName);
+            }
+        } catch (Exception e) {
+            throw new SubCommandException(this.getClass().getSimpleName() + " command failed", e);
+        } finally {
+            defaultMQAdminExt.shutdown();
+        }
+    }
+
+    private void printDiffForCluster(final DefaultMQAdminExt defaultMQAdminExt, String clusterName) throws SubCommandException {
+        ClusterInfo clusterInfo = null;
+        try {
+            clusterInfo = defaultMQAdminExt.examineBrokerClusterInfo();
+        } catch (Exception e) {
+            throw new SubCommandException(this.getClass().getSimpleName() + " command failed", e);
+        }
+
+        HashMap<String, Set<String>> clusterMap = clusterInfo.getClusterAddrTable();
+        Set<String> brokerNameSet = clusterMap.get(clusterName);
+        if (brokerNameSet == null || brokerNameSet.isEmpty()) {
+            System.out.printf("Can not find brokers for the cluster named %s!%n", clusterName);
+            return;
+        } else {
+            System.out.printf("%-24s %-24s %-24s %-14s%n",
+                    "#Cluster Name",
+                    "#Broker Name",
+                    "#Broker Address(Master)",
+                    "#Diff(Bytes)"
+            );
+            for (String brokerName : brokerNameSet) {
+                HashMap<String/* brokerName */, BrokerData> brokerAddrTable = clusterInfo.getBrokerAddrTable();
+                BrokerData brokerData;
+                if ((brokerData = brokerAddrTable.get(brokerName)) == null) {
+                    continue;
+                } else {
+                    String masterBrokerAddr = brokerData.getBrokerAddrs().get(MixAll.MASTER_ID);
+                    if (StringUtils.isBlank(masterBrokerAddr))
+                        continue;
+                    try {
```

Comment:
suggest all code in loop moving into try catch block.

Reply:
Thanks for reminding me. I've already optimized it.